### PR TITLE
[IIO] register iio source and add test cases

### PR DESCRIPTION
**Pull request by kparichay**
1. Register plugin as a nnstreamer plugin and initialize it
2. Add test case for setting/getting properties of tensor src iio
3. Bug fixes for the bugs found with the tests

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

**Self evaluation:**
1. Build test: [X ]Passed [ ]Failed [*]Skipped
2. Run test: [ X]Passed [ ]Failed [* ]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 1]
```diff
@@ -1612,10 +1612,18 @@ gst_tensor_src_iio_get_caps (GstBaseSrc * src, GstCaps * filter)
   GstTensorSrcIIO *self;
   GstCaps *caps;
   GstPad *pad;
+  GstPadTemplate *pad_template;
 
   self = GST_TENSOR_SRC_IIO (src);
   pad = self->srcpad;
-  caps = gst_pad_get_current_caps (pad);
+  if (GST_IS_PAD (pad)) {
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: [suggestion]
caps = gst_pad_get_current_caps (pad);
if (caps == NULL)
    caps = gst_pad_get_pad_template_caps (caps);
}

File: tests/meson.build [hunk 2]
```diff
@@ -86,6 +86,15 @@ if gtest_dep.found()
   )
 
   test('unittest_plugins', unittest_plugins, args: ['--gst-plugin-path=..'])
+
+  # Run unittest_src_iio
+  unittest_src_iio = executable('unittest_src_iio',
```
Comment (jaeyun-jung) [inline tests/meson.build]: [suggestion] 
to run testcase, insert tc into unittest_plugins
or update /debian/rules and /packaging/nnstreamer.spec

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 1 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: updated, 
also did a bug fix for the pads - 
_get_caps() can be called before _start()
earlier code set srcpad in properties in _start() and used it everywhere which is wrong
updated code to use basesrc->srcpad

File: tests/meson.build (hunk 2 above)
Comment (kparichay) [reply tests/meson.build]: updated debian/rules and packaging/nnstreamer.spec for testcases

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 3]
```diff
@@ -934,7 +935,7 @@ gst_tensor_src_iio_set_property (GObject * object, guint prop_id,
       break;
 
     case PROP_FREQUENCY:
-      self->sampling_frequency = g_value_get_uint64 (value);
+      self->sampling_frequency = (guint64) g_value_get_ulong (value);
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: ulong and uint64 are mixed up here.

```sampling_frequency``` is changed from ulong to uint64
and you are changing some related uint64 back to ulong. Make it consistent.

**Comment by myungjoo** [comment]
Other than the type consistency, all look good.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 3 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: sampling_frequency is interfaced as ulong for outside and stored as uint64 internally.
value is received from the user, which should be long format, hence the use of g_value_get_ulong(). 
as we store the value in uint64, we type cast it to uint64.

All the locations where the code interacts outside the file, ulong type is maintained.
Code had many inconsistencies which are handled by this commit.

Comment (myungjoo) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Oh,. ok. then. LGTM.

**Pull request merged**